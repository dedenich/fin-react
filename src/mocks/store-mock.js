import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const store = mockStore({
  admissions: [],
  remains: 0,
  collectedShare: 0,
  collectedAmount: 0,
  aim: {
    description: ``,
    amount: `0`,
  },
  isAimAdded: false,
});

export {store};
