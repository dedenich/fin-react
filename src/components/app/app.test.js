import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import {store as mockStore} from '../../mocks/store-mock.js';

import App from './app.jsx';

// const mockFn = jest.fn();

it(`App renders correctly`, () => {

  const tree = renderer
  .create(
      <Provider store={mockStore}>
        <App/>
      </Provider>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
