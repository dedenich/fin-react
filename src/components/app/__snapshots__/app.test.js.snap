// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="card aim-container"
    data-state="not empty"
    id="aim"
  >
    <div>
      <div
        className="aim-container__curr-cont"
      >
        <h2>
          12.10
        </h2>
        <div
          className="aim-container__cur-controls"
        >
          <div
            className="aim-container__controls_cur"
          >
            BYN
          </div>
        </div>
      </div>
      <div
        className="aim-container__aim-descr"
      >
        Билет в Витебск
      </div>
    </div>
  </div>,
  <div
    className="card-wrapper"
  >
    <p
      className="capture"
    >
      Поступления
      :
    </p>
    <div
      className="card"
    />
  </div>,
  <div
    className="card-wrapper"
  >
    <p
      className="capture"
    >
      Результат
      :
    </p>
    <div
      className="card"
    />
  </div>,
]
`;
